govcd/metadata_v2.go

```diff
@@ -0,0 +1,754 @@
+/*
+ * Copyright 2022 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"net/http"
+)
+
+// NOTE: This "v2" is not v2 in terms of API versioning, it's just a new way of handling metadata.
```

Comment:
This is my main question about this PR: do we really want "v2"? For the readers of the code, the "v2" will lead to wonder "Why v2? How it's different from v1? Etc.".

If yes, we should extend this comment with a sentence of how v2 is better than v1 :)

CC: @dataclouder @Didainius

Reply:
I like the fact of code being split. I just don't know a better name. IMO a good enough comment should be sufficient. That it supports metadata types, visibility, etc.

Reply:
I will revisit the comment and explain a bit better. I also thought about naming it `metadata_new.go` but I'm against using `new` suffix in most of the cases as its meaning is not very objective.

Reply:
Changed comment!

Reply:
It's great, thanks!